packages/server/src/time-control/time-control.ts

```diff
@@ -18,10 +24,26 @@ export function ValidateTimeControlConfig(
     time_control_config &&
     typeof time_control_config === "object" &&
     "type" in time_control_config &&
-    "mainTimeMS" in time_control_config
+    mainTimeMs in time_control_config &&
+    (time_control_config.type !== TimeControlType.Fischer ||
+      ValidateFischerConfig(time_control_config))
   );
 }
 
+const incrementMS: keyof IFischerConfig = "incrementMS";
+const maxTimeMsMS: keyof IFischerConfig = "maxTimeMS";
+/**
+ * Validates whether time_control_config has the additional
+ * properties for a Fischer time control.
+ */
+export function ValidateFischerConfig(time_control_config: object): boolean {
```

Comment:
Actually I would prefer not to use the typeguard after the config was validated in the game creation. If the server crashes because of it, it must be because of an implementation error, and we'd be made aware of it.
I have removed the todo for this reason.